core/src/data/value/string.rs

```diff
@@ -0,0 +1,133 @@
+use {
+    crate::{
+        data::{Literal, Value},
+        prelude::Key,
+    },
+    std::{borrow::Cow, cmp::Ordering},
+};
+
+impl PartialEq<String> for Value {
+    fn eq(&self, other: &String) -> bool {
+        match (self, other) {
+            (Value::Str(l), r) => l == r,
+            _ => false,
+        }
+    }
+}
+
+impl PartialOrd<String> for Value {
+    fn partial_cmp(&self, other: &String) -> Option<Ordering> {
+        match (self, other) {
+            (Value::Str(l), r) => Some(l.cmp(r)),
+            _ => None,
+        }
+    }
+}
+
+impl PartialEq<String> for Literal<'_> {
+    fn eq(&self, other: &String) -> bool {
+        match (self, other) {
+            (&Literal::Text(Cow::Borrowed(l)), r) => l == r,
+            _ => false,
+        }
+    }
+}
+
+impl PartialOrd<String> for Literal<'_> {
+    fn partial_cmp(&self, other: &String) -> Option<Ordering> {
+        match (self, other) {
+            (&Literal::Text(Cow::Borrowed(l)), r) => Some(l.cmp(r)),
+            _ => None,
+        }
+    }
+}
+
+impl From<&String> for Key {
+    fn from(s: &String) -> Self {
+        Key::Str(s.to_owned())
+    }
+}
```

Comment:
May be.. this would be too much to have `From` impl?
We can make String as Key with a single line of codes.